lib/rpmsg/rpmsg.c

```diff
@@ -185,7 +187,7 @@ static void rpmsg_unregister_endpoint(struct rpmsg_endpoint *ept)
 	rdev = ept->rdev;
 
 	if (ept->addr != RPMSG_ADDR_ANY)
-		rpmsg_release_address(rdev->bitmap, RPMSG_ADDR_BMP_SIZE,
+		rpmsg_release_address(rdev->bitmap, RPMSG_ADDR_BMP_BITS,
```

Comment:
Thanks to point the bug, but your patch seems not fix it. Right parameter here is RPMSG_ADDR_BMP_BITS that corresponds to the size of the bitmap table.
The issue is in rpmsg_is_address_set, rpmsg_set_address and rpmsg_release_address functions that should do the match conversion between the ept addr and the bit offset.

Reply:
Ok, I will move the conversion inside the function